sdk/nodejs/output.ts

```diff
@@ -131,6 +131,20 @@ class OutputImpl<T> implements OutputInstance<T> {
             promise: Promise<T>,
             isKnown: Promise<boolean>,
             isSecret: Promise<boolean>) {
+
+        isKnown = isKnown.then(known => {
+            if (!known) {
+               // If we were explicitly not-known, then we do not have to examine the value of the output at all.
+               // This is also desirable as it means if the value of the output is 'rejected' it doesn't affect the
+               // known/not-known promise for us.
+               return false;
+            }
+
+            // Otherwise, we are only known if the resolved value of the output contains no distinguished unknown
+            // values and if the value's promise is not rejected.
+            return promise.then(v => !containsUnknowns(v), () => false);
```

Comment:
I assume all `isKnown` promises are created by us, and so it's okay to silently swallow the error here?  Curious why we need that part of this - can `isKnown` ever become rejected?

Reply:
I don't think that it can ever become rejected in practice--this was all necessary in order to satisfy a test.